components/http/src/main/java/org/datacleaner/components/http/HttpRequestTransformer.java

```diff
@@ -0,0 +1,230 @@
+@Description("Sends a HTTP request for each record and retrieves the response as transformation output.")
+public class HttpRequestTransformer implements Transformer {
+
+    public static final String PROPERTY_INPUT_COLUMNS = "Input";
+    public static final String PROPERTY_VARIABLE_NAMES = "Variable names";
+    private static final String PROPERTY_URL = "URL";
+
+    @Inject
+    @Configured(value = PROPERTY_URL, order = 1)
+    @Description("The URL to invoke. The URL will be pre-processed by replacing any variable names in it with the corresponding dynamic values.")
+    String url = "http://";
+
+    @Inject
+    @Configured(order = 2)
+    HttpMethod method = HttpMethod.POST;
+
+    @Inject
+    @Configured(value = PROPERTY_INPUT_COLUMNS, order = 3)
+    InputColumn<?>[] input;
+
+    @Inject
+    @Configured(value = PROPERTY_VARIABLE_NAMES, order = 4)
+    @MappedProperty(PROPERTY_INPUT_COLUMNS)
+    String[] variableNames;
+
+    @Inject
+    @Configured(order = 5)
+    @StringProperty(multiline = true, emptyString = true)
+    @Description("The body of the request to invoke. The request body will be pre-processed by replacing any variable names in it with the corresponding dynamic values.")
+    String requestBody;
+
+    @Inject
+    @Configured(required = false, order = 100)
+    Map<String, String> headers;
+
+    @Inject
+    @Configured(required = false, order = 101)
+    String charset = HTTP.DEF_CONTENT_CHARSET.name();
+
+    @Inject
+    @Configured(required = false, order = 150)
+    @NumberProperty(negative = false, zero = false, positive = true)
+    @Description("The maximum number of requests that may be fired at the same time.\n"
+            + "Higher values may provide better throughput while it may also add load to the HTTP server.")
+    int maxConcurrentRequests = 20;
+
+    private CloseableHttpClient _httpClient;
+    private PooledServiceSession<Object[]> _session;
+
+    @Initialize
+    public void init() {
+        _httpClient = HttpClients.createSystem();
+        _session = new PooledServiceSession<>(maxConcurrentRequests);
+    }
+
+    @Close
+    public void close() {
+        FileHelper.safeClose(_httpClient, _session);
+    }
+
+    @Override
+    public OutputColumns getOutputColumns() {
+        final String[] columnNames = { "Response status code", "Response body" };
+        final Class<?>[] columnTypes = { Integer.class, String.class };
+        return new OutputColumns(columnNames, columnTypes);
+    }
+
+    @Override
+    public Object[] transform(InputRow inputRow) {
+        final Charset usedCharset = Charset.forName(charset);
+
+        final String requestBody = applyVariablesToString(this.requestBody, inputRow);
+        final String url = applyVariablesToString(this.url, inputRow);
+
+        final HttpUriRequest request = method.createRequest(url);
+        if (requestBody != null && request instanceof HttpEntityEnclosingRequest) {
```

Comment:
I prefer additional "(" and ")" to prevent errors in order of execution and enhances readibility.
So :
if ((requestBody != null)) && (..))